src/crypto/crypto_context.cc

```diff
@@ -191,25 +195,25 @@ int SSL_CTX_use_certificate_chain(SSL_CTX* ctx,
 }  // namespace
 
 X509_STORE* NewRootCertStore() {
-  static std::vector<X509*> root_certs_vector;
-  static Mutex root_certs_vector_mutex;
   Mutex::ScopedLock lock(root_certs_vector_mutex);
 
-  if (root_certs_vector.empty() &&
+  if (!root_certs_loaded &&
       per_process::cli_options->ssl_openssl_cert_store == false) {
     for (size_t i = 0; i < arraysize(root_certs); i++) {
-      X509* x509 =
+      X509Pointer x509 = X509Pointer(
           PEM_read_bio_X509(NodeBIO::NewFixed(root_certs[i],
                                               strlen(root_certs[i])).get(),
-                            nullptr,   // no re-use of X509 structure
+                            nullptr,    // no re-use of X509 structure
                             NoPasswordCallback,
-                            nullptr);  // no callback data
+                            nullptr));  // no callback data
 
       // Parse errors from the built-in roots are fatal.
       CHECK_NOT_NULL(x509);
 
-      root_certs_vector.push_back(x509);
+      root_certs_vector.push_back(std::move(x509));
```

Comment:
If `NODE_EXTRA_CA_CERTS` was specified at startup, the `roots_certs_vector` will not be empty at this point. This change causes the "extra CAs" to be loaded before the "bundled CAs", the opposite of today. My expectation is that this won't cause a problem since we wouldn't expect an extra CA to "spoof" a real one, but please review carefully in case I'm wrong. 😊